code/_core/obj/item/weapon/ranged/_ranged.dm

```diff
@@ -386,15 +386,15 @@ obj/item/weapon/ranged/proc/play_shoot_sounds(var/mob/caller,var/list/shoot_soun
 
 	return TRUE
 
-obj/item/weapon/ranged/proc/shoot(var/mob/caller,var/atom/object,location,params,var/damage_multiplier=1,var/click_called=FALSE)
+obj/item/weapon/ranged/proc/shoot(var/mob/caller,var/atom/object,location,params,var/damage_multiplier=1,var/click_called=FALSE,var/projectile_override = null)
 
 	if(!pre_shoot(caller,object,location,params,damage_multiplier))
 		return FALSE
 
 	var/quality_bonus = get_quality_bonus(0.5,2)
 	var/quality_penalty = max(1,1/get_quality_bonus(0.25,2))
 
-	var/obj/projectile/projectile_to_use = projectile
+	var/obj/projectile/projectile_to_use = (projectile_override != null && ispath(projectile_override, /obj/projectile) ? projectile_override : projectile)
```

Comment:
I would probably remove the ispath check here because, believe it or not, ispath and istype checks are expensive. If it causes errors because of a bad projectile_override var, then that's on the coder to fix.

Reply:
Noted for future reference, will remove.